### ISSUE TITLE
bundle size significantly larger for experimental vs legacy components

### ISSUE THREAD

**Issue by jponeill** [angular/components#27392]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

I know this is work in progress, but when I compare the bundle size of the legacy components vs the experimental ones, the experimental ones are significantly  larger (50% larger, in my project). I've attached source-map-explorer that shows the difference for form-field, button, dialog, expansion & icon.
<img width="513" alt="Screenshot 2023-06-30 at 11 54 29 AM" src="https://github.com/angular/components/assets/17234759/5c0a88a9-329e-4136-9655-41ca89f1b67a">
<img width="513" alt="Screenshot 2023-06-30 at 11 44 19 AM" src="https://github.com/angular/components/assets/17234759/7de692a0-f932-4b6f-863b-dbf3195ef634">

### Reproduction

StackBlitz link: 
Steps to reproduce:
1. Build project w/ material legacy components 
2.Build project w/ material experimental components
3. Compare bundle size using source-map-explorer
... experimental bundle size is much larger.

### Expected Behavior

I expect legacy and experimental bundle sizes to be close in size

### Actual Behavior

Experimental bundle sizes are much larger.

### Environment

- Angular: 16.2.0-next.0
- CDK/Material: 16.2.0-next.1
- Browser(s): N/A
- Operating System (e.g. Windows, macOS, Ubuntu): MacOS

**Comment by jponeill** [comment]
You're right, @crisbeto , the CSS bundle size is ~9K smaller. Accounting for that, it looks like there's ~40% increase for these components.

**Comment by jponeill** [comment]
@crisbeto I see that the legacy components have been removed in v17. I'm looking forward to switching over, but am a little worried about the increase in bundle size based on my quick tests.

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
